### PR TITLE
fix(metadata): warn devs when saving data longer than column holds

### LINKED ISSUE #10861: Add warning if metastring exceeded column length
Without `STRICT_TRANS_TABLES` MySQL truncates metastrings to fit the `text` column. This is problematic when metastring's integrity is important, e.g. a long serialized string. At least a warning would help debug issues that arise from that.

### PR DESCRIPTION
**Pull request by raghukul01**
Fixes #10861

### CODE REVIEW

**Comment by raghukul01** [comment]
screenshot 3
can you explain this error

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 1]
```diff
@@ -41,6 +41,8 @@ class MetadataTable {
 	
 	/** @var string */
 	protected $table;
+
```
Comment (mrclay) [inline engine/classes/Elgg/Database/MetadataTable.php]: This line has unneeded whitespace. Delete it.

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 2]
```diff
@@ -123,6 +125,10 @@ function create($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
 		}
 	
 		$access_id = (int) $access_id;
+		
+		if (strlen($value) > self::MYSQL_TEXT_BYTE_LIMIT) {
+ 			elgg_log("Metadata {$name} is above the MySQL TEXT size limit and may be truncated.", 'WARNING');
```
Comment (mrclay) [inline engine/classes/Elgg/Database/MetadataTable.php]: This line and a few others have a space at the beginning of the line. Remove them.

**Comment by raghukul01** [comment]
It again reports the same error

**Comment by beck24** [comment]
```
FILE: ...s/build/Elgg/Elgg/engine/classes/Elgg/Database/MetadataTable.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 127 | ERROR | [x] Tabs must be used to indent lines; spaces are not
     |       |     allowed
     |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 128 | ERROR | [x] Tabs must be used to indent lines; spaces are not
     |       |     allowed
     |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 215 | ERROR | [x] Tabs must be used to indent lines; spaces are not
     |       |     allowed
     |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 216 | ERROR | [x] Tabs must be used to indent lines; spaces are not
     |       |     allowed
     |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
```

The line numbers listed in the error message, these are lines you have modified.  They are not passing the code formatting policy because it's detecting spaces where there should be tabs.  Remove the spaces and use the tab indentation on lines 127, 128, 215, and 216.

http://learn.elgg.org/en/stable/contribute/code.html#php-guidelines

**Comment by beck24** [comment]
Don't give up, we appreciate the effort :)

**Comment by raghukul01** [comment]
@beck24 thank you sir for your help. I have made all the changes suggested by you all. Now I am creating a fresh new pull request with there changes, so I kindly request you to accept my PR.

**Review by beck24 — APPROVED** [review]

**Comment by jdalsem** [comment]
I think current state of the PR is fine, but it would be nice if there was a unittest made to verify this works

**Comment by beck24** [comment]
Normally I'd agree, but in this case I can't see how it could possibly not work, and @raghukul01 is new and has put in a *ton* of effort getting this far :)
I think we cut him some slack on the unit test front

**Comment by jdalsem** [comment]
Tests are partially meant to check if it works now, but also to check when changes happen in the future. Will merge it for now.

**Comment by jdalsem** [comment]
Used a 'squash and merge' commit to squash the 6 commits of this PR into 1

**Comment by raghukul01** [comment]
thank you very much to all of you. It is great working with you guys.
Thank you sir

**Pull request merged**